feat(web): support sortOption for ReactiveList

mohdashraf010897 opened the pull request:
**PR Type** 🏗️ `Enhancement`

**Description** Supports a `sortOption` key url-params to retain sort order for ReactiveList. https://github.com/appbaseio/reactivesearch/issues/261

[📔 Notion](https://www.notion.so/appbase/2a5bc1196ce6450793bd34f57ff6716b#625fdb9996d44a1caad3997ad345a524)

https://www.loom.com/share/ad2bcac00406439992ea9ccfc78fd7a0

mohdashraf010897 committed:
fix: resolve conflicts

mohdashraf010897 committed:
Merge branch 'next' of https://github.com/appbaseio/reactivesearch into next

mohdashraf010897 committed:
Merge branch 'next' of https://github.com/appbaseio/reactivesearch into next

mohdashraf010897 committed:
feat: support sortOption param for ReactiveList

mohdashraf010897 committed:
sync submodule

mohdashraf010897 committed:
fix: change logic for sortOption in url params

mohdashraf010897 committed:
fix: revert unneccessary changes

mohdashraf010897 committed:
fix: revert unneccessary changes

bietkul commented on packages/web/src/components/result/ReactiveList.js:
```diff
@@ -622,6 +614,18 @@ class ReactiveList extends Component {
 
 	updatePageURL = (page) => {
+		if (this.props.sortOptions) {
+			const sortOption = this.props.sortOptions[this.sortOptionIndex].label;
```
Add a `null` check before reading `label` key.

mohdashraf010897 replied:
Fixed here:  https://github.com/appbaseio/reactivesearch/pull/1863/commits/2dce811ce4202c2bad7b37632f0843b118a78dad

mohdashraf010897 committed:
fix(web): add null check before reading label

mohdashraf010897 commented:
@bietkul 
https://www.loom.com/share/ae934d6e678a4b8184dd12fe152079a1

mohdashraf010897 committed:
feat(vue): support for sortOptions

mohdashraf010897 commented:
Demo for [feat(vue): support for sortOptions](https://github.com/appbaseio/reactivesearch/pull/1863/commits/50477e641be718168034a900d235262a7811a3fc)
https://www.loom.com/share/2a22581d6caa4d928e512c435015a246

mohdashraf010897 committed:
sync reactivecore

bietkul commented:
@mohdashraf010897 The Vue demo that you shared doesn't reset the page on sort change.

mohdashraf010897 committed:
Merge branch 'next' of https://github.com/appbaseio/reactivesearch into next

mohdashraf010897 committed:
sync submodule

mohdashraf010897 committed:
fix(vue): reset page on changing sort option

mohdashraf010897 committed:
fix(vue): update basic example to use native pagination

mohdashraf010897 commented:
> @mohdashraf010897 The Vue demo that you shared doesn't reset the page on sort change.

Updated demo 
https://www.loom.com/share/71f665ef5de242d98ec536b36774d31c

also updated the basic example
